conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/EndpointDetails.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.undertow.lib;
+
+import com.palantir.tokens.auth.ImmutablesStyle;
+import io.undertow.util.HttpString;
+import org.immutables.value.Value;
+
+@Value.Immutable
+@ImmutablesStyle
+public interface EndpointDetails {
```

Comment:
I'd prefer a concrete class here, no reason to allow folks to subclass

Reply:
Sure, but what is the risk of people subclassing it ? This is easier to mock this way.